zapcore/sampler.go

```diff
@@ -100,13 +170,16 @@ type sampler struct {
 // Keep in mind that zap's sampling implementation is optimized for speed over
 // absolute precision; under load, each tick may be slightly over- or
 // under-sampled.
+//
+// Deprecated: use NewSamplerWithOptions.
 func NewSampler(core Core, tick time.Duration, first, thereafter int) Core {
 	return &sampler{
 		Core:       core,
 		tick:       tick,
 		counts:     newCounters(),
 		first:      uint64(first),
 		thereafter: uint64(thereafter),
+		hook:       NopSamplingHook,
 	}
```

Comment:
We should make it so that the old implementation calls the new one so that if we make changes to `sample` initialization, they don't have to go in two places.

```suggestion
	return NewSamplerWithOptions(core, tick, first, thereafter)
```

Reply:
Good call.